aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -71,7 +71,7 @@ Tensor values_sparse(const Tensor& self) {
 /*** Helper methods ***/
 
 SparseTensor new_sparse(const TensorOptions& options) {
-  AT_ASSERT(!options.is_variable());  // TODO: remove this when Variable and Tensor are merged
+  TORCH_INTERNAL_ASSERT(impl::variable_is_excluded());
```

Comment:
is it true that these statements are equivalent?  I believe that's true for `tensor.is_variable()` and `impl::variable_is_excluded()` but I'm not sure about `options.is_variable()`.

Reply:
No, they're not equivalent. `TensorOptions::is_variable` doesn't consult the TLS at call site; there is a value stored in TensorOptions. I don't think that value is seeded from TLS either; it was manually twiddled before (all that code is gone now).

I think the new code makes sense without appealing to the old code. The analogy is with the other factory functions, which also assert variable is excluded.